lib/src/Base/Stat/DiracCovarianceModel.cxx

```diff
@@ -234,6 +234,11 @@ struct DiracCovarianceModelDiscretizePolicy
 
 CovarianceMatrix DiracCovarianceModel::discretize(const Sample & vertices) const
 {
+  // If we have to check for repeated points and there are repeated points, use the generic discretization method
+  if (ResourceMap::GetAsBool("DiracCovarianceModel-CheckUnique") && (vertices.sortUnique().getSize() != vertices.getSize()))
```

Comment:
This means that each call of `discretize`/`discretizeAndFactorize` requires:
 - Checking the resource map value
 - Making a sortUnique & compare the sizes

I would rather rewrite the algo like this: 
```cxx
if (!ResourceMap::GetAsBool("DiracCovarianceModel-CheckUnique")
{
  // implement custom code; It is user responsibility
return discretizeUnique(vertices);
}
// Now we need to check
if (vertices.sortUnique().getSize() != vertices.getSize())  
  return CovarianceModelImplementation::discretize(vertices);
return discretizeUnique(vertices);
```
The purpose is to not `sort` the vertices & compute its unique values at each optim iteration

Reply:
Thanks to have taken the time to read the code.

The C++ language, as well as the C language, imposes to implement shortcuts in tests. If `ResourceMap::GetAsBool("DiracCovarianceModel-CheckUnique")` is false, `(vertices.sortUnique().getSize() != vertices.getSize())` is not evaluated at all. So the implementation proposed in this PR has the exact same cost as the one you propose.

Reply:
Ok I see the point, thanks